AIFG-3D-Connect-4/FourTechRulesHandler.cpp

```diff
@@ -16,30 +16,10 @@ FourTechRulesHandler::FourTechRulesHandler(GameBoard & t_board) :
 ///////////////////////////////////////////////////////////////////////////////
 void FourTechRulesHandler::update()
 {
-	Coordinate pos{ 0, 0, 0 };
-
-	if (m_playersTurn)
-	{
-		// Takes the player's input - will be moved to it's own class with time.
-		std::cout << "Enter your move: ";
-
-		std::cin >> pos.x >> pos.y >> pos.z;
-
-		if (PieceType::None != m_board.getPiece(pos))
-		{
-			std::cout << "Invalid position" << std::endl;
-			std::cin.ignore(std::numeric_limits<std::streamsize>::max());
-			std::cin.get();
-			return;
-		}
-	}
-	else
-	{
-		pos = m_ai.getMove();
-	}
-
+	// Gets the position either the player or AI wants.
+	Coordinate pos = m_turnHandler.getCoordinate();
 	// Get the piece type and place it.
-	PieceType type = m_playersTurn ? PieceType::Red : PieceType::Yellow;
+	PieceType type = m_turnHandler.getPlayersTurn() ? PieceType::Red : PieceType::Yellow;
```

Comment:
I would replace this and line 20 with a getMove() function from the turn handler which gets the coordinates from the input and creates a Move object with the coordinate and PieceType.